pinot-common/src/main/java/org/apache/pinot/common/messages/SegmentReloadMessage.java

```diff
@@ -34,19 +34,22 @@ public class SegmentReloadMessage extends Message {
   public static final String RELOAD_SEGMENT_MSG_SUB_TYPE = "RELOAD_SEGMENT";
 
   private static final String FORCE_DOWNLOAD_KEY = "forceDownload";
+  private static final String SEGMENT_NAMES = "segmentNames";
 
-  public SegmentReloadMessage(@Nonnull String tableNameWithType, @Nullable String segmentName, boolean forceDownload) {
+  public SegmentReloadMessage(String tableNameWithType, List<String> segmentNames, boolean forceDownload) {
```

Comment:
Shall we make `segmentNames` nullable and never empty? Passing empty list to reload all segments can cause confusion. Also not sure if `getSegmentList()` will return `null` in that case

Reply:
works for me. updated.